src/transaction_builder.js

```diff
@@ -462,15 +467,17 @@ TransactionBuilder.prototype.__build = function (allowIncomplete) {
   this.inputs.forEach(function (input, i) {
     var scriptType = input.redeemScriptType || input.prevOutType
     if (!scriptType && !allowIncomplete) throw new Error('Transaction is not complete')
-
     var result = buildInput(input, allowIncomplete)
 
     // skip if no result
-    if (!result && allowIncomplete) return
-    if (result && result.script) return tx.setInputScript(i, result.script)
-    if (result && result.witness) return tx.setWitness(i, result.witness)
+    if (!allowIncomplete) {
+      if (SIGNABLE.indexOf(result.type) === -1 && result.type !== bscript.types.P2WPKH) {
+        throw new Error(result.type + ' not supported')
+      }
```

Comment:
Man,  I know we both agree the complexity explosion in this area has always sucked, and made it so hard to review changes by any one.

Its all just so gross.